.github/workflows/ci.yml

```diff
@@ -3,49 +3,112 @@ name: CI
 on:
   push:
     branches:
+      - develop
       - master
   pull_request:
     branches:
-      - master
       - develop
+      - master
 
 jobs:
   linux:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        host: ["x86_64-pc-linux-gnu", "x86_64-w64-mingw32"]
-    env:
-      NO_QT: 1
     steps:
-      - uses: actions/checkout@v1
-      - name: install dependencies
-        run: |
-          sudo add-apt-repository ppa:bitcoin/bitcoin
-          sudo apt-get install -y build-essential libtool autotools-dev automake pkg-config bsdmainutils curl git python3
-          sudo apt-get install -y libevent-dev libboost-system-dev libboost-filesystem-dev libboost-test-dev libboost-thread-dev libminiupnpc-dev libzmq3-dev libdb4.8-dev libdb4.8++-dev
-      - if: matrix.host == 'x86_64-w64-mingw32'
-        name: mingw preinstall script
+      - name: checkout
+        uses: actions/checkout@v1
+        with:
+          path: main
+      - name: Prepare ccache timestamp
+        id: cache_timestamp
         run: |
-          sudo apt-get install -y g++-mingw-w64-x86-64
-          sudo update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
-          sudo update-alternatives --install /usr/bin/x86_64-w64-mingw32-gcc x86_64-w64-mingw32-gcc /usr/bin/gcc 90
-          sudo update-alternatives --install /usr/bin/x86_64-w64-mingw32-g++ x86_64-w64-mingw32-g++ /usr/bin/g++ 90
+          current_date=`date --utc +%Y%m%d_%H%M%SZ`
+          echo "::set-output name=timestamp::${current_date}"
+      - name: ccache cache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: ~/.ccache
+          key: linux-ccache-${{ steps.cache_timestamp.outputs.timestamp }}
```

Comment:
do you think timestamp is good idea? 
given 2 builds, their timestamps will be different -> no cache will be reused

Reply:
Cache is re-used during my tests, so this should be no issue.  It restores based on name, following other examples I included the timestamp, but it has no effect as to it being used..